packages/config/src/forkConfig/index.ts

```diff
@@ -138,15 +138,32 @@ export function createForkConfig(config: ChainConfig): ForkConfig {
       }
       return sszTypesFor(forkName);
     },
-    getMaxBlobsPerBlock(fork: ForkName): number {
-      switch (fork) {
-        case ForkName.fulu:
-          return config.MAX_BLOBS_PER_BLOCK_FULU;
-        case ForkName.electra:
-          return config.MAX_BLOBS_PER_BLOCK_ELECTRA;
-        default:
-          return config.MAX_BLOBS_PER_BLOCK;
+    getMaxBlobsPerBlock(epoch: Epoch): number {
+      if (config.BLOB_SCHEDULE.length === 0) {
+        throw Error("Attempt to get MAX_BLOBS_PER_BLOCK from empty BLOB_SCHEDULE");
       }
+
+      const minEpoch = Math.min(...config.BLOB_SCHEDULE.map((e) => e.EPOCH));
```

Comment:
@ensi321 I removed all checks based on fork and only used epoch, we can either check for min epoch like this or we assert against `DENEB_FORK_EPOCH` like done [here](https://github.com/ethereum/consensus-specs/pull/4277/files#diff-d2350537413777662eb3c51d3ff4cd400ca832866071f05b9f44203b80a9aa90R43) in the spec

Reply:
> @ensi321 I removed all checks based on fork and only used epoch, we can either check for min epoch like this or we assert against `DENEB_FORK_EPOCH` like done [here](https://github.com/ethereum/consensus-specs/pull/4277/files#diff-d2350537413777662eb3c51d3ff4cd400ca832866071f05b9f44203b80a9aa90R43) in the spec

This change does not look right.

In the case we have a devnet
```
network_params:
  fulu_fork_epoch: 1
  bpo_1_epoch: 1
  bpo_1_max_blobs: 9
  bpo_2_epoch: 2
  bpo_2_max_blobs: 18
```
According to the spec, this config is valid because every epoch in the blob schedule is >= `DENEB_FORK_EPOCH`, and max blob should be 9 during Electra epoch (epoch 0). So we should not throw error here, and should return a default value (minimum of max blob in blob schedule)

Reply:
seems we should just fall through and return lowest value
```python
def get_max_blobs_per_block(epoch: Epoch) -> uint64:
    """
    Return the maximum number of blobs that can be included in a block for a given epoch.
    """
    assert len(BLOB_SCHEDULE) > 0
    for entry in sorted(BLOB_SCHEDULE, key=lambda e: e["EPOCH"], reverse=True):
        if epoch >= entry["EPOCH"]:
            return entry["MAX_BLOBS_PER_BLOCK"]
    return min(entry["MAX_BLOBS_PER_BLOCK"] for entry in BLOB_SCHEDULE) # <-- here
```

Reply:
I have updated it in https://github.com/ChainSafe/lodestar/pull/7729/commits/7c36eb1c0f4dec0de6ae662c92980dc9ec58e2f6 to follow spec more closely, but still, this seems like a fallback that is pretty sketchy, it should never happen on a actual network, only for testing, but ideally the BPO schedule you shared should define max blobs for epoch 0 since genesis is post deneb

Reply:
> or we assert against DENEB_FORK_EPOCH like done [here](https://github.com/ethereum/consensus-specs/pull/4277/files#diff-d2350537413777662eb3c51d3ff4cd400ca832866071f05b9f44203b80a9aa90R43) in the spec

don't think we need to assert this, the spec function doesn't do it either, there is no network anymore that as is pre-deneb